jqm-all/jqm-engine/src/main/java/com/enioka/jqm/tools/JmxSslHandshakeListener.java

```diff
@@ -0,0 +1,221 @@
+/**
+ * Copyright © 2013 enioka. All rights reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.enioka.jqm.tools;
+
+import java.security.Principal;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.net.ssl.HandshakeCompletedEvent;
+import javax.net.ssl.HandshakeCompletedListener;
+import javax.security.auth.login.LoginException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class JmxSslHandshakeListener implements HandshakeCompletedListener
+{
+    private static JmxSslHandshakeListener instance;
+    private Logger jqmlogger = LoggerFactory.getLogger(JmxSslHandshakeListener.class);
+
+    /**
+     * Time in milliseconds during which a SSL handshake success is saved for an
+     * user. <br>
+     * During this time, considering an user who attempted to connect with SSL
+     * client authentication providing the certificate of an existing user named
+     * USERNAME (Common Name of the certificate), any client certificate can be used
+     * to authenticate with the (USERNAME, password of USERNAME) credentials,
+     * because it doesn't seem that we have a mean to know which client certificate
+     * corresponds to the user processed in {@link JmxLoginModule}.
+     */
+    private static final long SSL_SUCCESS_EXPIRE_TIME = 3000L;
+
+    /**
+     * Time in milliseconds during which all SSL handshake successes of an user
```

Comment:
Please keep documentation and comments concise.

Some description here may concern the class itself, and should move to an appropriate location.